testnet/stacks-node/src/main.rs

```diff
@@ -55,9 +55,9 @@ use backtrace::Backtrace;
 
 fn main() {
     panic::set_hook(Box::new(|panic_info| {
-        eprintln!("Process abort due to thread panic: {}", panic_info);
+        error!("Process abort due to thread panic: {}", panic_info);
```

Comment:
So, the reason this was `eprintln!` was because I wanted some level of guarantee that the `panic_info` and backtrace would get logged even if the logger was somehow broken or blocked on I/O (or even memory pressure).

Reply:
Yes -- in the case of a JSON logger, any JSON-expecting consumer will lose this information without creating special case logic for panics. This is the case for Hiro's nodes, and would be the case for most people that just try to capture the logs using standard tools.

The stacks node's term logger writes to `stderr` and flushes on every macro invocation, so it would log before this panic finished (same as `eprintln!`) -- the only difference is if there is not enough memory to buffer this log line, the process would get killed before it had a chance to log the panic message, but that is exceedingly unlikely: the panic would need to occur at a time when there's just the log message (perhaps 100kb) worth of memory left to the process. This wouldn't be the case in a normal OOM situation: the panic handler won't get invoked then at all.